integ/workflow/headers/routers.go

```diff
@@ -0,0 +1,92 @@
+package headers
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/indeedeng/iwf/gen/iwfidl"
+	"github.com/indeedeng/iwf/service"
+	"log"
+	"net/http"
+)
+
+const (
+	WorkflowType = "headers"
+	State1       = "S1"
+)
+
+type handler struct {
+	invokeHistory map[string]int64
+}
+
+func NewHandler() *handler {
+	return &handler{
+		invokeHistory: make(map[string]int64),
+	}
+}
+
+// ApiV1WorkflowStartPost - for a workflow
+func (h *handler) ApiV1WorkflowStateStart(c *gin.Context) {
+	headerValue := c.GetHeader(TestHeaderKey)
+	if headerValue != TestHeaderValue {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "test header not found"})
```

Comment:
I think we need to return here after `c.JSON`. Otherwise the remaining code will override it.